### ISSUE TITLE
Investigate ligcc and lig++

### ISSUE THREAD

**Issue by probonopd** [probonopd/AppImageKit#49]
According to http://listaller.tenstral.net/docs/chap-Listaller-ManualPages.html#idp15415040 ligcc is a wrapper around gcc. It allows you to create more portable executables by doing three things:

 * Forces the linker to link against older glibc symbols. Users who are using an older version of glibc will no longer get "undefined symbol GLIBC_2.4 in /lib/libc.so"-style error messages.
 * Allows you to easily statically link to any other library.
 * Automatically removes bogus dependencies. For example, your program uses libfoo. libfoo uses libpng internally, but your app does not. Yet the pkg-config file for libfoo specifies "-lfoo -lpng" is linker parameters. And tadaa - you now have a bogus dependency on libpng! LiGCC automatically removes the -lpng for you if your app doesn't use libpng directly.
 * Add $ORIGIN/../lib to the binary's library search path. $ORIGIN is the directory in which the binary exists. This ensures that your binary can find library dependencies which are placed in the 'lib' folder under the same prefix. You might want to install custom libraries into /lib. If you set $APBUILD_PROJECTNAME, ligcc will also add $ORIGIN/../lib/$APBUILD_PROJECTNAME to the library search path.

**Comment by darealshinji** [comment]
> Automatically removes bogus dependencies. For example, your program uses libfoo. libfoo uses libpng internally, but your app does not. Yet the pkg-config file for libfoo specifies "-lfoo -lpng" is linker parameters. And tadaa - you now have a bogus dependency on libpng! LiGCC automatically removes the -lpng for you if your app doesn't use libpng directly.

This can be done by adding `-Wl,--as-needed` to the LDFLAGs. On Ubuntu GCC is patched to do that by default.

**Comment by TheAssassin** [comment]
Is this issue still relevant?

**Comment by probonopd** [comment]
Yes, I think we should investigate this. It may help us to produce more compatible build products.
